### ISSUE TITLE
OpenShift Projects and ProjectRequests not ordered appropriately with namespaces

### ISSUE THREAD

**Issue by tylerauerbeck** [kubernetes-sigs/kustomize#2563]
When creating a kustomization that creates Projects/ProjectRequests, kustomize doesn't create these objects in the order that I'd expect. Since Projects/ProjectRequests are functionally namespaces, I would expect them to be created as some of the initial objects just as namespaces are. However it appears that kustomize treats them more like CustomResources and doesn't try to create them until the end of the list (which causes issues for any objects that need to be created in those projects.

Is there anyway (ideally in v2.0.3 which is bundled with `oc` and `kubectl`) to force this order?

**Comment by mpreu** [comment]
The simplest solution is to use the `reorder` option.

```bash
kustomize build --reorder none ...
```
Using that option you are on your own how to order objects. I use that in combination with OCP successfully, but currently in a 3.11 cluster. Therefore I cannot verify if that option is included in the latest `oc` bundle. If that's not the case I assume the internal ordering is enforced and only k8s resources are considered. No way around it from my pov.

**Comment by tylerauerbeck** [comment]
Thanks @mpreu -- that's what I had thought the case was going to be, but would also like to hear from some of the folks on the kustomize end to see if they've got input as well. 

I was hoping to be able to stick with features that are available in `v2.0.3` just so that I would be able to run a single command `oc apply -k ...`, but looks like that may not be feasible at this point. I thought I saw some hack around making OpenShift specific objects look like CRD's (which I would assume might get them added toward the top of the list, but also don't know if that's a great approach either.

**Comment by mpreu** [comment]
@tylerauerbeck Fair enough. 
Just to give some final notes as I tried something similar before `reorder none` was introduced. 

You are maybe referring to something like this https://github.com/kubernetes-sigs/kustomize/issues/97 and this https://github.com/adnovum/kustomize-openshift (supporting CRDs before `transformer configurations` offered an easier way).

Unfortunately at the time I tried, this method did not allow for "rewriting" the `kind` of a resources, so `kind` comparison is still happening to the hard-coded list (looking at the code). Would be interesting to see if one can get that actually to work, but in the end it would be more of an academic example.

Unfortunate to see that the CLIs are not integrating a more recent version https://github.com/kubernetes/kubectl/issues/818.
